### PR TITLE
[FEAT] Spreadsheet export service

### PR DESCRIPTION
**Pull request by laurensmertens**
|Q              |A  |
|---------------|---|
|Bug fix?       |No|
|New feature?   |Yes|	
|BC breaks?     |No|	
|Deprecations?  |No|	
|Fixed tickets  |No|	

Service to allow spreadsheet exports based on JSON.

### CODE REVIEW

File: src/Common/SpreadsheetGeneratorService.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    public function generateSpreadsheet(array $config): void
+    {
+        $defaults = self::getDefaults();
+
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults($defaults);
+        $resolver->setRequired(['writer', 'filename', 'sheets']);
+        $resolver->setAllowedValues('writer', ['xlsx']);
+
+        $resolver->resolve($config);
```
Comment (theus77) [inline src/Common/SpreadsheetGeneratorService.php:26]: I would isolate it in a function with the right return hint

File: tests/Unit/Common/SpreadsheetGeneratorTest.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace EMS\CommonBundle\Tests\Unit\Common;
+
+use EMS\CommonBundle\Common\SpreadsheetGeneratorService;
+use PHPUnit\Framework\TestCase;
+
+class SpreadsheetGeneratorTest extends TestCase
+{
+    /** @var SpreadsheetGeneratorService */
+    private $spreadSheetGenerator;
```
[the remark this answers is not in the document]
Comment (laurensmertens) [reply tests/Unit/Common/SpreadsheetGeneratorTest.php:11]: It's not yet available in PHP 7.2

[the remark this answers is not in the document]
Comment (theus77) [reply tests/Unit/Common/SpreadsheetGeneratorTest.php:11]: Ok. I'll upgrade it

**Review by theus77 — APPROVED** [review]

**Comment by theus77** [comment]
I let David approve and merge it

File: src/Common/SpreadsheetGeneratorService.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+        $spreadsheet = $this->buildUpSheets($config);
+
+        $writer = new Xlsx($spreadsheet);
+        \header('Content-Type: application/vnd.openxmlformats-officedocument.spreadsheetml.sheet');
+        \header('Content-Disposition: attachment; filename="'.\urlencode($config['filename'].'.xlsx').'"');
+        $writer->save('php://output');
+        exit;
```
[the remark this answers is not in the document]
Comment (Davidmattei) [reply src/Common/SpreadsheetGeneratorService.php]: +1

**Review by Davidmattei — CHANGES_REQUESTED** [review]

**Pull request merged**